src/components/[guild]/Requirements/components/GuildCheckout/hooks/useAllowSpendingToken.ts

```diff
@@ -0,0 +1,27 @@
+import { MaxUint256 } from "@ethersproject/constants"
+import useContract from "hooks/useContract"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import useSubmit from "hooks/useSubmit"
+import ERC20_ABI from "static/abis/erc20Abi.json"
+import useAllowance from "./useAllowance"
+
+const useAllowSpendingTokens = (tokenAddress: string, contract: string) => {
```

Comment:
I think it'd be nicer for useAllowance to contain this logic too, there's no need for a separate hook